test/app-luatest/tarantool_module_no_internal_fields_test.lua

```diff
@@ -0,0 +1,33 @@
+local it = require('test.interactive_tarantool')
+
+local t = require('luatest')
+local g = t.group()
+
+g.before_each(function()
+    g.child = it.new()
+end)
+
+g.after_each(function()
+    if g.child ~= nil then
+        g.child:close()
+    end
+end)
+
+-- Verify that no internal fields are shown for the 'tarantool'
+-- module table at serialization to console.
+g.test_no_internal_fields = function()
+    -- Print content of the module table.
+    g.child:execute_command("require('tarantool')")
+    local response = g.child:read_response()
+
+    -- Collect fields that start from the underscore.
+    local internal_keys = {}
+    for k, _ in pairs(response) do
+        if k:startswith('_') then
+            table.insert(internal_keys, k)
+        end
+    end
+
+    -- Ensure that there are no such fields.
+    t.assert_equals(internal_keys, {})
```

Comment:
Minor: I'd rather check that `next(internal_keys)` yields `nil`, since I'm not sure whether `assert_equals` compares tables by values (I've already checked the comment [here](https://github.com/tarantool/luatest/blob/master/luatest/assertions.lua#L173)). Feel free to ignore.

Reply:
I would do this in a project's code, but I like the luatest's deep comparison assertion due to its visual simplicity and verbose error reporting (when the check fails). Given way ideally fits for testing.